File: 0206-reverse-linked-list/0206-reverse-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 prev = None
        curr = head
        while curr:
            next_temp = curr.next
            curr.next = prev
            prev = curr
            curr = next_temp
            
        return prev
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        
        ListNode curr = head;
        ListNode prev = null;
        
        while (curr!=null){
            ListNode next = curr.next;
            curr.next = prev;
            prev = curr;
            curr = next;
        
        }
        return prev;
        
    
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        """
        3 pointer approach
        1 for prev
        1 for curr
        1 for next
        """

        prev = None
        next = None

        while head:
            next = head.next
            head.next = prev
            prev = head
            head = next

        return prev

